eos-updater/poll.c

```diff
@@ -211,7 +218,10 @@ get_finders (SourcesConfig          *config,
         }
     }
 
-  g_ptr_array_add (finders, NULL);  /* NULL terminator */
+  if (offline_finders->len > 0)
+    g_ptr_array_add (offline_finders, NULL);  /* NULL terminator */
```

Comment:
Why's the NULL terminator needed? These are `GPtrArray`s which know their own length.

Reply:
Ah, because their `pdata` ends up being passed as `OstreeRepoFinder **` to `ostree_repo_find_remotes_async`.